modules/va_facilities/lib/va_facilities/param_validators.rb

```diff
@@ -34,8 +41,104 @@ def validate_type_and_services_known
+    end
+
+    def validate_lat
+      Float(params[:lat]) if params[:lat]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('lat', params[:lat])
+    end
+
+    def validate_lng
+      Float(params[:lng]) if params[:lng]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('lng', params[:lng])
+    end
+
+    def validate_bbox
+      raise ArgumentError unless params[:bbox].nil? || params[:bbox]&.length == 4
+      params[:bbox]&.each { |x| Float(x) }
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('bbox', params[:bbox])
+    end
+
+    def validate_a_param_exists(require_one_param)
+      if none_required_params_exists?(require_one_param, params)
+        raise Common::Exceptions::ParameterMissing.new(require_one_param.first.to_s,
+                                                       detail: MISSING_FACILITIES_PARAMS_ERR)
+      end
+    end
+
+    def valid_location_query?
+      case location_keys
+      when [] then true
+      when %i[lat long] then true
+      when [:state]     then true
+      when [:zip]       then true
+      when [:bbox]      then true
+      else
+        # There can only be one
+        render json: {
+          errors: ['You may only use ONE of these distance query parameter sets: lat/long, zip, state, or bbox']
+        },
+               status: 422
+      end
+    end
+
+    def validate_required_nearby_params(required_params)
+      param_keys = params.keys.map(&:to_sym)
+      address_params = required_params[:address]
+      lat_lng_params = required_params[:lat_lng]
+      address_difference = (address_params - param_keys)
+      lat_lng_difference = (lat_lng_params - param_keys)
+
+      unless valid_params?(address_difference, lat_lng_difference)
+        if ambiguous?(address_difference, lat_lng_difference, address_params, lat_lng_params)
+          raise Common::Exceptions::AmbiguousRequest.new(detail: AMBIGUOUS_PARAMS_ERR)
+        elsif address_difference != address_params
+          raise Common::Exceptions::ParameterMissing.new(address_difference.to_s, detail: MISSING_NEARBY_PARAMS_ERR)
+        elsif lat_lng_difference != lat_lng_params
+          raise Common::Exceptions::ParameterMissing.new(lat_lng_difference.to_s, detail: MISSING_NEARBY_PARAMS_ERR)
+        end
+      end
+    end
+
+    private
+
+    def none_required_params_exists?(required_one_param, params)
+      lat_lng = %i[lat long]
+      lat_lng_exists = params.key?(:lat) && params.key?(:long)
+      required_params = required_one_param - lat_lng
+      !lat_lng_exists && required_params.none? { |param| params.key? param }
+    end
+
+    def valid_params?(difference1, difference2)
+      (difference1.empty? && difference2.any?) || (difference2.empty? && difference1.any?)
+    end
+
+    def ambiguous?(address_difference, lat_lng_difference, address_params, lat_lng_params)
+      address_difference.empty? || lat_lng_difference.empty? ||
```

Comment:
If I'm understanding the logic correctly, the `ambiguous?` method only works right now because it is wrapped in the `unless valid_params?` conditional.  The following change would make it work independently of that conditional.

Is this the logic?:
It is ambiguous If (both parameter sets are empty) OR if (both parameter sets have values present but are incomplete)

```suggestion
      (address_difference.empty? && lat_lng_difference.empty?) ||
```

Reply:
oops you are correct, changed it